pkg/compactor/bucket_compactor.go

```diff
@@ -814,13 +836,31 @@ func (c *BucketCompactor) Compact(ctx context.Context, maxCompactionTime time.Du
 							c.bkt,
 							err.(OutOfOrderChunksError).id,
 							block.OutOfOrderChunksNoCompactReason,
-							"OutofOrderChunk: marking block with out-of-order series/chunks to as no compact to unblock compaction", c.metrics.blocksMarkedForNoCompact); err == nil {
+							"OutofOrderChunk: marking block with out-of-order series/chunks to as no compact to unblock compaction", c.metrics.blocksMarkedForNoCompact.WithLabelValues(block.OutOfOrderChunksNoCompactReason)); err == nil {
 							mtx.Lock()
 							finishedAllJobs = false
 							mtx.Unlock()
 							continue
 						}
 					}
+
+					// In case an unhealthy block is found, we mark it for no compaction
+					// to unblock future compaction run.
+					if IsCriticalError(err) && c.skipUnhealthyBlocks {
+						if err := block.MarkForNoCompact(
```

Comment:
nit: can we move call to `block.MarkForNoCompact` outside of `if` statement to make reading this code easier? (On first look it wasn't clear to me why we call `mtx.Lock()` in the function parameters, before I scrolled to the right to see what's actually happening)